app/src/main/java/com/github/libretube/util/NowPlayingNotification.kt

```diff
@@ -166,72 +162,161 @@ class NowPlayingNotification(
-                    handlePlayerAction(PREV)
-                    return SessionResult.RESULT_SUCCESS
+            override fun onCustomAction(action: String, extras: Bundle?) {
+                handlePlayerAction(action)
+                super.onCustomAction(action, extras)
+            }
+        }
+
+        val playbackState = if (player.isPlaying) {
+            createPlaybackState(PlaybackStateCompat.STATE_PLAYING)
+        } else {
+            createPlaybackState(PlaybackStateCompat.STATE_PAUSED)
+        }
+
+        mediaSession = MediaSessionCompat(context, TAG())
+        mediaSession.setCallback(sessionCallback)
+        mediaSession.setPlaybackState(playbackState)
+        mediaSession.setMetadata(getMetadataFromPlayer(player.mediaMetadata))
+
+        val playerStateListener = object : Player.Listener {
+            override fun onIsPlayingChanged(isPlaying: Boolean) {
+                super.onIsPlayingChanged(isPlaying)
+
+                val newPlaybackState = if (isPlaying) {
+                    createPlaybackState(PlaybackStateCompat.STATE_PLAYING)
+                } else {
+                    createPlaybackState(PlaybackStateCompat.STATE_PAUSED)
+                }
+
+                mediaSession.setPlaybackState(newPlaybackState)
+                mediaSession.setMetadata(getMetadataFromPlayer(player.mediaMetadata))
+            }
+
+            override fun onIsLoadingChanged(isLoading: Boolean) {
+                super.onIsLoadingChanged(isLoading)
+
+                if (!isLoading) {
+                    mediaSession.setMetadata(getMetadataFromPlayer(player.mediaMetadata))
                 }
-                return super.onPlayerCommandRequest(session, controller, playerCommand)
             }
 
-            override fun onPostConnect(
-                session: MediaSession,
-                controller: MediaSession.ControllerInfo
-            ) {
-                session.setCustomLayout(getCustomActions())
+            override fun onMediaMetadataChanged(mediaMetadata: MediaMetadata) {
+                super.onMediaMetadataChanged(mediaMetadata)
+                mediaSession.setMetadata(getMetadataFromPlayer(mediaMetadata))
             }
         }
 
-        mediaSession = MediaSession.Builder(context, player)
-            .setCallback(sessionCallback)
-            .build()
-        mediaSession.setCustomLayout(getCustomActions())
+        player.addListener(playerStateListener)
     }
 
-    private fun getCustomActions() = mutableListOf(
-        // disabled and overwritten in onPlayerCommandRequest
-        // createMediaSessionAction(R.drawable.ic_prev_outlined, PREV),
-        createMediaSessionAction(R.drawable.ic_next_outlined, NEXT),
-        createMediaSessionAction(R.drawable.ic_rewind_md, REWIND),
-        createMediaSessionAction(R.drawable.ic_forward_md, FORWARD)
-    )
+    private fun getMetadataFromPlayer(metadata: MediaMetadata): MediaMetadataCompat {
```

Comment:
This should be an extension function for MediaMetadata:
```kotlin
fun MediaMetada.toMediaMetadataCompat(): MediaMetadataCompat {
...
}
```
in the `extensions` package.